cli/src/pcluster/templates/pcluster_dashboard_builder.py

```diff
@@ -0,0 +1,586 @@
+        )
+
+        # Create a text widget for title "Head Node EC2 metrics"
+        self._add_text_widget("\n# Head Node EC2 Metrics\n", cloudwatch_dashboard)
+
+        # Head node instance graphs
+        self._add_head_node_instance_metrics_graphs(cloudwatch_dashboard)
+
+        ebs_metrics = [
+            {"metrics": ["VolumeReadOps", "VolumeWriteOps"], "title": "Read/Write Ops"},
+            {"metrics": ["VolumeReadBytes", "VolumeWriteBytes"], "title": "Read/Write Bytes"},
+            {"metrics": ["VolumeTotalReadTime", "VolumeTotalWriteTime"], "title": "Total Read/Write Time"},
+            {"metrics": ["VolumeQueueLength"], "title": "Queue Length"},
+            {"metrics": ["VolumeIdleTime"], "title": "Idle Time"},
+        ]
+        conditional_ebs_metrics = [
+            {
+                "metric": "VolumeConsumedReadWriteOps",
+                "supported_vol_types": ["io1", "io2", "gp3"],
+                "title": "Consumed Read/Write Ops",
+            },
+            {
+                "metric": "VolumeThroughputPercentage",
+                "supported_vol_types": ["io1", "io2", "gp3"],
+                "title": "Throughput Percentage",
+            },
+            {"metric": "BurstBalance", "supported_vol_types": ["gp2", "st1", "sc1"], "title": "Burst Balance"},
+        ]
+
+        # EBS metric graphs
+        number_of_ebs_volumes = len(self._storage_resource_ids[SharedStorageType.EBS])
+        if number_of_ebs_volumes > 0:
+            self._add_ebs_metrics_graphs(
+                cloudwatch_dashboard=cloudwatch_dashboard,
+                ebs_metrics=ebs_metrics,
+                conditional_ebs_metrics=conditional_ebs_metrics,
+            )
+
+        # Raid metric graphs
+        number_of_raid_volumes = len(self._storage_resource_ids[SharedStorageType.RAID])
+        if number_of_raid_volumes > 0:
+            self._add_raid_metrics_graphs(
+                cloudwatch_dashboard=cloudwatch_dashboard,
+                ebs_metrics=ebs_metrics,
+                conditional_ebs_metrics=conditional_ebs_metrics,
+            )
+
+        # EFS metrics graphs
+
+        number_of_efs_volumes = len(self._storage_resource_ids[SharedStorageType.EFS])
+        if number_of_efs_volumes > 0:
+            self._add_efs_metrics_graphs(cloudwatch_dashboard)
+
+        # FSx metrics graphs
+        number_of_fsx_volumes = len(self._storage_resource_ids[SharedStorageType.FSX])
+        if number_of_fsx_volumes > 0:
+            self._add_fsx_metrics_graphs(cloudwatch_dashboard)
+
+        # Head Node logs, if CW Logs are enabled
+        if self.cw_logging_enabled:
+            self._add_cw_log(cloudwatch_dashboard)
+
+    def _update_coord(self, coord, dx, dy):
+        """Calculate coordinates for the new graph. It updates both x and y values if the graph doesn't fit in the
+        given line."""
+        coord.update({"x": coord["x"] + dx})
```

Comment:
You can avoid passing coord to all the methods because you have it as class attribute.
You initialized it with:
`self.coord` so you can simply use it in the methods.